workbench-core/datasets/src/dataSetService.ts

```diff
@@ -50,16 +52,33 @@ export class DataSetService {
    * @returns the DataSet object which is stored in the backing datastore.
    */
   public async provisionDataSet(request: CreateProvisionDatasetRequest): Promise<DataSet> {
-    const { storageProvider, ...dataSet } = request;
-
-    await storageProvider.createStorage(dataSet.storageName, dataSet.path);
-
-    const provisioned: DataSet = {
-      ...dataSet,
-      storageType: storageProvider.getStorageType()
+    const metadata: Metadata = {
+      actor: { id: '-', roles: [] },
+      action: this.provisionDataSet.name,
+      source: {
+        serviceName: DataSetService.name
+      },
+      requestBody: request
     };
 
-    return await this._dbProvider.addDataSet(provisioned);
+    try {
+      const { storageProvider, ...dataSet } = request;
+
+      await storageProvider.createStorage(dataSet.storageName, dataSet.path);
+
+      const provisioned: DataSet = {
+        ...dataSet,
+        storageType: storageProvider.getStorageType()
+      };
+      const response: DataSet = await this._dbProvider.addDataSet(provisioned);
+      metadata.statusCode = 200;
+      await this._audit.write(metadata, response);
+      return response;
+    } catch (error) {
+      metadata.statusCode = 400;
```

Comment:
Why are we hard-coding this to be a 400? Could it ever be another 4xx or 5xx error?

Reply:
see previous comment. Expect this to go away.